modules/ROOT/pages/custom-ingress-configuration.adoc

```diff
@@ -0,0 +1,52 @@
+= Defining a Custom Ingress Configuration
+
+[NOTE]
+====
+The tasks outline in this topic must be performed by an IT administrator
+====
+
+Runtime Fabric on Self-managed Kubernetes enables you to edit a template file that specifies a custom Ingress configuration. Using this file you can define annotations, rewrite rules, the ingressClassName, and more.
+
+The template file is a Kubernetes ConfigMap file. After customizing the template, you can use the `rtfctl apply` command-line utility to apply the custom Ingress configuration to your Kubernetes cluster.
+
+If you do not apply a custom template file, Runtime Fabric uses a default configuration that works with the out-of-the-box Kubernetes Ingress configuration.
+
+== Procedure
+
+. Copy the following into a text file:
++
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ingress-config
+  namespace: rtf
+data:
+  ingress.annotations: |
+    nginx.ingress.kubernetes.io/rewrite-target: /
+    nginx.ingress.kubernetes.io/ssl-redirect: true
+    kubernetes.io/ingress.class: <name>
+  ingress.spec: |
+    ingressClassName: <name>
+    rules.http.paths.path: /${APP_PATH}
+----
+
+. Edit the text file as required.
++
+All changes should be made under the `data` section of this file. Do not modify any other sections of the template. The most common configuration options are.
++
+* `ingress.annotations`: Annotations defined in this section are added to the `annotations` section of every Ingress resource created of a Mule application. Each annotation must be defined on a new line. Annotations defined here are default annotations that you can override.
+
+* `ingress.spec`: Used to configure the resource `spec` paramaters
+** `ingressClassName` (optional): Specifies the class name used in the Ingress resource created for a Mule application. Note, this is supported only on Kubernetes version 1.18 and higher.
```

Comment:
Also we should mention that its user responsibility to make sure that they are using the version of ingress controller which has support for this ingressClassName K8 feature.
Ex: I have seen few HAProxy community editions which still relies on `kubernetes.io/ingress.class: <name>` annotation even on k8 1.18 and higher clusters. 

So from our template side we support these users adding annotation and still getting it to work. 
cc: @kmendiratta-crm

Reply:
We can clarify it better. how about this::
`ingressClassName` (optional): Specifies the class name used in the Ingress resource created for a Mule application. 
Note: ingressClassName within ingress spec is supported on Kubernetes version 1.18 and higher. Few ingress controllers may still use the annotations to define the ingressClassName.
Note: Ingress class name should be specified in ingress.annotations for Kubernetes version 1.17 and below i .e kubernetes.io/ingress.class: <ingress-class-name>

Reply:
Yes, this looks perfect